File: fieldmate/src/main/java/com/nguyencong/fieldmate/service/RuleService.java
package com.nguyencong.fieldmate.service;

import com.nguyencong.fieldmate.dto.request.RuleRequest;
import com.nguyencong.fieldmate.dto.response.RuleResponse;

public interface RuleService {

    RuleResponse createRule(Long venueId, RuleRequest request);
    RuleResponse updateRule(Long id, RuleRequest request);
    void deleteRule(Long id);
}

File: front-end/fieldmate-web/src/services/rule.service.ts
import {
  fieldmateClient,
  fieldmateEndpoints,
} from "@/configs/fieldmate-client";
import type { RuleRequest, RuleResponse } from "@/types/rule";

export const ruleService = {
  async create(venueId: number, data: RuleRequest) {
    const response = await fieldmateClient.post<RuleResponse>(
      fieldmateEndpoints.secureVenueRules(venueId),
      data,
    );

    return response.data;
  },

  async update(ruleId: number, data: RuleRequest) {
    const response = await fieldmateClient.put<RuleResponse>(
      fieldmateEndpoints.secureRule(ruleId),
      data,
    );

    return response.data;
  },

  async remove(ruleId: number) {
    await fieldmateClient.delete(fieldmateEndpoints.secureRule(ruleId));
  },
};
